plugins/modules/zos_replace.py

```diff
@@ -0,0 +1,646 @@
+                        break
+                    else:
+                        search_before = True
+                        search_after = False
+
+            elif pattern_begin.match(line) is not None:
+                begin_block_code = line_counter + 1
+                match = True
+                if not before:
+                    break
+                else:
+                    search_before = True
+                    search_after = False
+
+        if search_before:
+            if lit_bf:
+                if line.find(before) > -1:
+                    end_block_code = line_counter
+                    match = True
+                    break
+            elif pattern_end.match(line) is not None:
+                end_block_code = line_counter
+                match = True
+                break
+
+        line_counter += 1
+
+    return begin_block_code, end_block_code, match
+
+
+def open_file(file, encoding, uss):
+    """
+    Args
+    ----------
+        file (_type_): _description_
+        encoding (_type_): _description_
+        uss (_type_): _description_
+
+    Returns
+    ----------
+        _type_: _description_
+    """
+    decode_list = []
+
+    if uss:
+        with io.open(file, "rb") as content_file:
+            content = codecs.decode(content_file.read(), encoding)
+        decode_list = content.splitlines()
+    else:
+        with zoau_io.RecordIO(f"//'{file}'") as dataset_read:
+            dataset_content = dataset_read.readrecords()
+        decode_list = [codecs.decode(record, encoding) for record in dataset_content]
+
+    return decode_list
+
+
+def replace_func(file, regexp, replace, module, uss, literal, encoding="cp1047", after="", before=""):
+    """Function to extract from the uss a fragment or the full text to be replaced and replace the content.
+
+    Args
+    ----------
+        file : str
+            Uss path.
+        regexp : str
+            Regex expression to search.
+        replace : str
+            Regex expression or text to replace.
+        module : obj
+            Object of Ansible to access to the facilities
+        uss : bool
+            Variable to indicate the type of open for the module
+        literal : list
+            List of values to be used as string disabling regex option.
+        encoding : str, optional
+            Encoding to use en decoding content.
+            Defaults to "cp1047".
+        after : str, optional
+            Str or regex to search where to start the section to replace on the text.
+            Defaults to "".
+        before : str, optional
+            Str or regex to search where to end the section to replace on the text.
+            Defaults to "".
+
+    Returns
+    ----------
+        list : List with the new text with the replace expected.
+    """
+    decode_list = open_file(file, encoding, uss)
+
+    lit_rex = False
+
+    if len(literal) > 0:
+        lit_rex = True if "regexp" in literal else False
```

Comment:
I would change this variable name to:
```suggestion
        is_regex = True if "regexp" in literal else False
```